File: server/features/browser.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// BrowserCDPStream describes the CDP stream HTTP handler for embedded browser sessions.
var BrowserCDPStream = featureregistry.Feature{
	ID:          "browser-cdp-stream",
	Title:       "CDP Stream",
	Description: "Streams Chrome DevTools Protocol events for a session's embedded browser.",
	RPCIDs:      []string{"browser:cdp-stream"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// BrowserProxy describes the VNC proxy HTTP handler for embedded browser sessions.
var BrowserProxy = featureregistry.Feature{
	ID:          "browser-proxy",
	Title:       "Browser Proxy",
	Description: "Proxies VNC connections to the embedded browser for a session.",
	RPCIDs:      []string{"browser:proxy"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(BrowserCDPStream)
	featureregistry.Register(BrowserProxy)
}

File: web-app/src/lib/features/features/browser.ts
import type { Feature } from '../types';

export const browserFeatures = {
  'browser-cdp-stream': {
    id: 'browser-cdp-stream',
    title: 'CDP Stream',
    description: 'Streams Chrome DevTools Protocol events for a session\'s embedded browser.',
    rpcIds: ['browser:cdp-stream'],
    componentPaths: [],
    testIds: [
      'TestCDPStreamHandler_MissingSessionID_Returns400',
      'TestCDPStreamHandler_UnknownSessionID_Returns404',
      'TestCDPStreamHandler_CDPPortZero_Returns503',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'browser-proxy': {
    id: 'browser-proxy',
    title: 'Browser Proxy',
    description: 'Proxies VNC connections to the embedded browser for a session.',
    rpcIds: ['browser:proxy'],
    componentPaths: [],
    testIds: [
      'TestVNCProxyHandler_MissingSessionID_Returns400',
      'TestVNCProxyHandler_UnknownSessionID_Returns404',
      'TestVNCProxyHandler_VNCPortZero_Returns503',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;
